src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -1522,6 +1522,25 @@ public function save(array $options = [])
     }
 
     /**
+     * Save the model to the database using transaction.
+     *
+     * @param  array  $options
+     * @return bool
+     *
+     * @throws \Throwable
+     */
+    public function saveOrFail(array $options = [])
+    {
+        $saved = false;
+
+        $this->getConnection()->transaction(function () {
+            $saved = $this->save($options);
```

Comment:
this function will not modify the saved variable - php puts it in a different scope